src/com/SocketCommunication.cpp

```diff
@@ -134,9 +134,15 @@ void SocketCommunication::acceptConnectionAsServer(std::string const &acceptorNa
                                                    int                requesterCommunicatorSize)
 {
   PRECICE_TRACE(acceptorName, requesterName, acceptorRank, requesterCommunicatorSize);
-  PRECICE_CHECK(requesterCommunicatorSize > 0, "Requester communicator size has to be > 0!");
+  PRECICE_CHECK(requesterCommunicatorSize >= 0, "Requester communicator size has to be positve!");
```

Comment:
```suggestion
  PRECICE_ASSERT(requesterCommunicatorSize >= 0, "Requester communicator size has to be positve.");
```

Reply:
This is called on each rank, thus this assertion must hold.

Reply:
I am not sure if I understand what you mean. You mean that this should indeed be a `CHECK` and no `ASSERT` ?

Reply:
The requester size always has to be `>0` as at least the Master calls this.
The requester size has to be checked by the remote SolverInterface, thus this should never be triggered. Therefore I think this should be an assertion.

Reply:
Yes, so the suggestion is good, right? :smirk:

Reply:
Sorry! I misread the diff! :sweat_smile: 
Yes the suggestion is correct.